Deadline submission updates

RyanLloydZA opened the pull request:
Preliminary Updates before the main template change updates

Fixes bug on handling compositions submitted starting from frame Zero.
Updates to using the current backup file on the farm rather than the users working scene. (This will be updated again with the templates update)

Updates Job Naming on deadline for clarity and to keep it inline with the new standard for nuke

RyanLloydZA committed:
Fix conditional check for display start time to ensure proper handling of None values, also fixes a bug that comps starting from 0 would bypass the frame list calculation and would default to the standard 1000-1500 frame range

RyanLloydZA committed:
Updating job naming conventions for clarity. Moved Render submissions to use deadline backup file

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -1437,7 +1440,7 @@ def submit_to_deadline(self, save_project=True):
             self.message_box("Deadline Submission", "Submission completed successfully.\n\n%d jobs submitted to Deadline." % num_successful_submissions)
 
-    def submit_render_queue_item_to_deadline(self, render_queue_item, deadline_settings, project_path, layers, previous_job_id, version):
+    def submit_render_queue_item_to_deadline(self, render_queue_item, deadline_settings, project_path, layers, previous_job_id, version, render_file):
```
Pretty minor thing, but this line is getting quite long. 

With functions with lots of args, it's a little easier to read when you format it like this:
```
def submit_render_queue_item_to_deadline(
    self, 
    render_queue_item, 
    deadline_settings, 
    project_path, 
    layers, 
    previous_job_id, 
    version, 
    render_file
    ):
```

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -1628,7 +1633,7 @@ def apply_and_submit(self):
             self,
             "Save Project",
-            "Do you want to save the project before submitting to Deadline?",
+            "Do you want to save the project before submitting to Deadline? \n\nIt's recommended to save the project to ensure whats being submitted is up to date.",
```
This line is also long... I always struggle with how format long strings like this. 

I think this is decent:
```
save_project = QtGui.QMessageBox.question(
    self,
    "Save Project",
    "Do you want to save the project before submitting to Deadline? \n\n" 
    "It's recommended to save the project to ensure whats being submitted is up to date.",
    ...
```

But then it looks like an additional argument 🤔 Maybe this instead then?
```
save_project = QtGui.QMessageBox.question(
    self,
    "Save Project",
    "Do you want to save the project before submitting to Deadline? \n\n" \
    "It's recommended to save the project to ensure whats being submitted is up to date.",
    ...
```
(With the backslash added to highlight the fact it's a line continuation)

dianaflorea-territory replied:
Would it be worth forcing the save?

JacobTerritory replied:
This is a good call - If they are submitting to deadline, then they 100% should save the scene first. 

But un-prompted saves are also bad, so it should be a save/cancel type thing.

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -1569,7 +1573,8 @@ def submit_render_queue_item_to_deadline(self, render_queue_item, deadline_setti
         with open(plugin_info_path, "w") as plugin_info:
 
-            plugin_info.write(f"SceneFile={project_path}\n")
+            #plugin_info.write(f"SceneFile={project_path}\n"
```
you mentioned already on a call, but would be good to clean up this old line of code. 

(Can go in the next round of changes)

RyanLloydZA committed:
Added render button to dialog and implemented rendering of current queue items, enhancing user interface and functionality for local rendering.

Also Implemented copying of the current scene used for rendering to a templated location that will accompany the renders for publish. This copied scene is alo used with deadline submissions instead of the main working file

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -615,6 +587,136 @@ def apply_to_render_queue_items(self):
         logger.debug("Total Time: %s" % (time.time() - self.start_time))
         logger.debug("Total Render Queue Items Updated: %s" % count)
 
+    def process_queue_items_for_render(self):
+        """
+            Process the render queue items for rendering locally
+            Updates the status of the render queue items in the table and creates a copy
+            of the project file for each render queue item to be rendered into the templated location.
+
+            :returns: True if successful, False otherwise
+        """
```
Get out of here with this filthy doc string format!

(As an aside from this... My vs code has started auto completing doc strings in this format for me. I wonder if it's configurable?)

JacobTerritory replied:
And I'm not particularly attached to the google style format. Just consistency is important. 

Interestingly though, if we want to do automated docs, this style might actually be better because it has a slightly stricter syntax which means parsing will be more reliable.

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -615,6 +587,136 @@ def apply_to_render_queue_items(self):
+        if not project_check:
+            return False
+
+        # If there are rows in the table
```
"are no rows"

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -615,6 +587,136 @@ def apply_to_render_queue_items(self):
+                compName = comp.name
+                templateName = renderFormatDropdown.currentText()
+                render_queue_template = self.get_render_queue_template(row)
+                frame_range = self.get_frame_range(comp, row)
```
This appears to not be used.

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -615,6 +587,136 @@ def apply_to_render_queue_items(self):
+                render_queue_item = tableItem.data(QtCore.Qt.UserRole)
+                comp = render_queue_item.comp
+                compName = comp.name
+                templateName = renderFormatDropdown.currentText()
```
This appears to not be used.

JacobTerritory commented on python/app/dialog.py:
```diff
@@ -802,22 +952,38 @@ def message_box(self, title, text):
         if any(template_file_name.startswith(fmt) for fmt in self.render_preset_movie_formats):
-            templateName = self._app.get_setting("mov_render_template")
+            template_map = {
+                (True, True): "comp_render_scene_template",
+                (True, False): "mov_render_comp_template",
+                (False, True): "render_scene_template",
+                (False, False): "mov_render_template",
+            }
         else:
-            templateName = self._app.get_setting("seq_render_template")
+            template_map = {
+                (True, True): "comp_render_scene_template",
+                (True, False): "seq_render_comp_template",
+                (False, True): "render_scene_template",
+                (False, False): "seq_render_template",
+            }
+
+        templateName = self._app.get_setting(template_map[(use_comp_name, render_scene)])
```
This is really weird, but also clever and elegant. I like it!

JacobTerritory replied:
I wonder if we can simplify the templates so that we don't need so many different options. 

But Idk what the templates look like at all for this, so maybe it's not possible.

RyanLloydZA committed:
Updated Doc Strings, CLeaned up some unused variables.

There are still a few more to cover but commit current progress

RyanLloydZA committed:
Enhance dialog UI with animated toggle panel and side quick actions menu; added refresh button and improved button styling.

Updated Doc Strings
